clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -990,8 +990,14 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
   }
 
   // Allocate static local memory in SYCL kernel scope for each allocation call.
-  if (LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice) {
+    // Group local memory pass depends on inlining. Turn it on even in case if
```

Comment:
If I understand correctly, always inliner already works when optimizations are disabled, see in this file:
 ```
665   if (CodeGenOpts.OptimizationLevel <= 1) {
 666     bool InsertLifetimeIntrinsics = ((CodeGenOpts.OptimizationLevel != 0 &&
 667                                       !CodeGenOpts.DisableLifetimeMarkers) ||
 668                                      LangOpts.Coroutines);
 669     PMBuilder.Inliner = createAlwaysInlinerLegacyPass(InsertLifetimeIntrinsics);
 670   } else {

```

Reply:
@mlychkov, I suggest you to take a look at [clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp](https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp) LIT test - this is the way you should be able to write a unit-test for the change, plus you can double-check the information from Dmitry at the same time

Reply:
> If I understand correctly, always inliner already works when optimizations are disabled, see in this file:
> 
> ```
> 665   if (CodeGenOpts.OptimizationLevel <= 1) {
> 666     bool InsertLifetimeIntrinsics = ((CodeGenOpts.OptimizationLevel != 0 &&
> 667                                       !CodeGenOpts.DisableLifetimeMarkers) ||
> 668                                      LangOpts.Coroutines);
> 669     PMBuilder.Inliner = createAlwaysInlinerLegacyPass(InsertLifetimeIntrinsics);
> 670   } else {
> ```

Please, notice check at the beginning of that function:
```
636 void EmitAssemblyHelper::CreatePasses(legacy::PassManager &MPM,
637                                       legacy::FunctionPassManager &FPM) {
638   // Handle disabling of all LLVM passes, where we want to preserve the
639   // internal module before any optimization.
640   if (CodeGenOpts.DisableLLVMPasses)
641     return;
```

Reply:
@AlexeySachkov Thank you for the tip.
Added FE test for the fix.